### PR TITLE
Dat 4227 = menu navigation: explore wiki clickability test

### PR DESCRIPTION
**Pull request by nikodamn**
Description
Additional test for navigation menu - when clicked 'explore XXX wiki' user is redirected to that wiki's main page.

Reviewers
@Wikia/west-wing

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java [hunk 1]
```diff
@@ -173,6 +187,15 @@ public boolean isLogoutLinkVisible() {
     }
   }
 
+  public boolean isExploreWikiHeaderVisible() {
+    try {
+      return exploreWikiHeader.isDisplayed();
+    } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java [hunk 2]
```diff
@@ -78,6 +81,17 @@ public Navigation clickBackButton() {
     return this;
   }
 
+  public Navigation clickExploreWikiHeader() {
+    Loading loading = new Loading(driver);
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java]: MAJOR Rename "loading" which hides the field declared at line 58. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AHiddenFieldCheck)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java [hunk 3]
```diff
@@ -95,12 +111,26 @@ public void mercury_navigation_navigationElementsUserLoggedIn() {
...
 
-    topBar.openNavigation();
     Assertion.assertTrue(navigation.isUserAvatarVisible());
     Assertion.assertFalse(navigation.isUserProfileLinkVisible());
     Assertion.assertFalse(navigation.isLogoutLinkVisible());
     Assertion.assertEquals(navigation.getNavigationHeaderText(), "Sign In | Register");
   }
+
+  @Test(groups = "mercury_navigation_exploreWikiNavigatesToWikiMainPage")
+  public void mercury_navigation_exploreWikiNavigatesToWikiMainPage() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java [hunk 4]
```diff
@@ -42,22 +38,30 @@ public void mercury_topbar_topBarIsAlwaysVisible() {
 
   @Test(groups = "mercury_topbar_closeButtonAppears")
   public void mercury_topbar_closeButtonAppears() {
-    init();
+    TopBar topBar =
+        new ArticlePage()
+            .open(MercurySubpages.MAIN_PAGE)
+            .getTopBar();;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java]: MAJOR Remove this empty statement. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues:
* MAJOR 4 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java [hunk 5]
```diff
@@ -70,25 +74,35 @@ public void mercury_topbar_switchBetweenSearchAndNavigation() {
...
 
   @Test(groups = "mercury_topbar_wikiaLogoRedirectsToFandomPage")
   public void mercury_topbar_wikiaLogoRedirectsToFandomPage() {
-    init();
+    TopBar topBar =
+        new ArticlePage()
+            .open(MercurySubpages.MAIN_PAGE)
+            .getTopBar();
 
     Assertion.assertTrue(topBar.isLogoVisible());
```
Comment (SebastianMarzjan) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java]: I think if it's not visible, click should fail?

Comment (nikodamn) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java]: Good catch, it's not necessary here.

File: src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java [hunk 6]
```diff
@@ -173,6 +187,15 @@ public boolean isLogoutLinkVisible() {
     }
   }
 
+  public boolean isExploreWikiHeaderVisible() {
+    try {
+      return exploreWikiHeader.isDisplayed();
```
Comment (rybmat) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java]: AFAIR this logic is duplicated in some other places. Did you consider to extract it somewhere and pass the element as an argument?

File: src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java [hunk 7]
```diff
@@ -174,6 +188,15 @@ public boolean isLogoutLinkVisible() {
...
+    } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java [hunk 8]
```diff
@@ -79,6 +82,17 @@ public Navigation clickBackButton() {
     return this;
   }
 
+  public Navigation clickExploreWikiHeader() {
+    Loading loading = new Loading(driver);
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java]: MAJOR Rename "loading" which hides the field declared at line 59. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AHiddenFieldCheck)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java (hunk 3 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java (hunk 6 above)
Comment (SebastianMarzjan) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java]: @ludwikkazmierczak - what would be the most appropriate place given the current architectural direction for selenium tests?

Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java]: IMHO - yes, `Navigation` seems like a good place.

File: src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java (hunk 8 above)
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java]: Just use `loading` declared class field

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java (hunk 3 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java [hunk 9]
```diff
@@ -9,42 +9,40 @@
 import com.wikia.webdriver.common.core.helpers.Emulator;
 import com.wikia.webdriver.common.core.helpers.User;
 import com.wikia.webdriver.common.templates.NewTestTemplate;
-import com.wikia.webdriver.elements.common.Navigate;
 import com.wikia.webdriver.elements.mercury.components.Navigation;
 import com.wikia.webdriver.elements.mercury.components.TopBar;
+import com.wikia.webdriver.elements.mercury.pages.ArticlePage;
+
+import org.testng.annotations.BeforeMethod;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java]: MINOR Remove this unused import 'org.testng.annotations.BeforeMethod'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java [hunk 10]
```diff
@@ -95,12 +91,28 @@ public void mercury_navigation_navigationElementsUserLoggedIn() {
...
 
-    topBar.openNavigation();
     Assertion.assertTrue(navigation.isUserAvatarVisible());
     Assertion.assertFalse(navigation.isUserProfileLinkVisible());
     Assertion.assertFalse(navigation.isLogoutLinkVisible());
     Assertion.assertEquals(navigation.getNavigationHeaderText(), "Sign In | Register");
   }
+
+  @Test(groups = "mercury_navigation_exploreWikiNavigatesToWikiMainPage")
+  public void mercury_navigation_exploreWikiNavigatesToWikiMainPage() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java (hunk 3 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by SebastianMarzjan** [comment]
LGTM

**Pull request merged**